src/test/client/TestClient.h

```diff
@@ -86,6 +93,128 @@ class ClientScaffold : public Client {
       wait_on_list(waiting_for_reclaim);
       return session->reclaim_state == MetaSession::RECLAIM_FAIL ? true : false;
     }
+
+    void write_n_bytes_async(struct Fh *fh, size_t to_write, size_t block_size,
+                            int iov_cnt, off_t *offset) {
+      /// @brief Write N bytes of data asynchronously.
+      /// @param fh - File handle
+      /// @param to_write - bytes to write
+      /// @param block_size - Total size of each iovec structs array
+      /// @param iov_cnt - Number of elements in iovec structs array
+      /// @param offset - location to start writing from
+
+      const size_t DATA_PER_BLOCK = size_t(block_size / iov_cnt);
+      // to persist memory address of out_buf after every loop iteration
+      std::vector<std::unique_ptr<char[]>> buffers;
```

Comment:
Why not just use `vector<std::string>`?

Reply:
in the for loop below, i'm storing `auto out_buf = std::make_unique<char[]>(DATA_PER_BLOCK);`, hence this.

Reply:
I *can* make use of `std::string` which would need me to make few changes the way i get a raw ptr in `iov_buf_out[i].iov_base = out_buf.get();` where https://cplusplus.com/reference/string/string/c_str/ would be helpful. Do you want me to make this change?

Reply:
I mean, we don't care of performance in the test case., so switching b/w `char *` and `std::string` should be fine.